src/quacc/calculators/espresso/utils.py

```diff
@@ -43,3 +46,39 @@ def get_pseudopotential_info(
             ecutrho = pp_dict[element]["cutoff_rho"]
         pseudopotentials[element] = pp_dict[element]["filename"]
     return ecutwfc, ecutrho, pseudopotentials
+
+
+def parse_ph_patterns(root_dir: str | Path, prefix: str) -> dict[int, int]:
```

Comment:
This function isn't used anywhere anymore, right? If so, please remove the function plus the imports introduced by this change.